commands/menu/message/rate_meme.py

```diff
@@ -0,0 +1,61 @@
+import hikari
+import lightbulb
+from behaviours import meme_rater
+from random import choice
+
+plugin = lightbulb.Plugin("meme_context_commands")
+
+loading_messages = [
+    "Rating meme...",
+    "Processing meme...",
+    "Analyzing meme...",
+    "Evaluating meme...",
+    "Assessing meme...",
+    "Grading meme...",
+    "Reviewing meme...",
+    "Examining meme...",
+    "Judging meme...",
+    "Scoring meme...",
+    "Stand back, meme rating in progress...",
+    "Hold tight, meme rating underway...",
+    "Just a moment, meme rating in progress...",
+]
+
+
+@plugin.command
+@lightbulb.command("Rate Meme", "Rate this meme using AI")
+@lightbulb.implements(lightbulb.MessageCommand)
+async def rate_meme_command(ctx: lightbulb.MessageContext) -> None:
+    """Rate a meme using the meme rater."""
+    message = ctx.options.target
+
+    # Check if message has any media content
+    if not message.attachments and not message.embeds:
```

Comment:
Also check if the message has already been rated.

Reply:
Previously the attachments were rated on message creation, and the embeds rated on message update and the results combined together. Now we're rating both the embeds and attachments on update, so there's no need to combine - the new result can just overwrite the previous.

However, that's not what this comment is about - this is just that there should be a `if already_rated: return`

Reply:
To check if it was previously rated, I checked the existence of emojis.
So it returns early, but this prevents returning rating + explanation to the command.

I think in this case fetching it from DB & returning it should do :thinking:

Reply:
Rate meme first fetched from db if rating is assigned. If yes, returns values from db.